test/common/router/router_ratelimit_test.cc

```diff
@@ -668,6 +668,187 @@ TEST_F(RateLimitPolicyEntryTest, DynamicMetaDataNonStringMatch) {
+      prop: foo
+  )EOF";
+
+  TestUtility::loadFromYaml(metadata_yaml, route_.metadata_);
+
+  rate_limit_entry_->populateDescriptors(route_, descriptors_, "", header_, default_remote_address_,
+                                         dynamic_metadata_);
+
+  EXPECT_THAT(std::vector<Envoy::RateLimit::Descriptor>({{{{"fake_key", "fake_value"}}}}),
+              testing::ContainerEq(descriptors_));
+}
+
+TEST_F(RateLimitPolicyEntryTest, RouteEnryMetaDataNoMatch) {
+  const std::string yaml = R"EOF(
+actions:
+- route_entry_metadata:
+    descriptor_key: fake_key
+    metadata_key:
+      key: 'envoy.xxx'
+      path:
+      - key: test
+      - key: prop
+  )EOF";
+
+  setupTest(yaml);
+
+  std::string metadata_yaml = R"EOF(
+filter_metadata:
+  envoy.xxx:
+    another_key:
+      prop: foo
+  )EOF";
+
+  TestUtility::loadFromYaml(metadata_yaml, route_.metadata_);
+
+  rate_limit_entry_->populateDescriptors(route_, descriptors_, "", header_, default_remote_address_,
+                                         dynamic_metadata_);
+
+  EXPECT_TRUE(descriptors_.empty());
+}
+
+TEST_F(RateLimitPolicyEntryTest, RouteEnryMetaDataEmptyValue) {
+  const std::string yaml = R"EOF(
+actions:
+- route_entry_metadata:
+    descriptor_key: fake_key
+    metadata_key:
+      key: 'envoy.xxx'
+      path:
+      - key: test
+      - key: prop
+  )EOF";
+
+  setupTest(yaml);
+
+  std::string metadata_yaml = R"EOF(
+filter_metadata:
+  envoy.xxx:
+    test:
+      prop: ""
+  )EOF";
+
+  TestUtility::loadFromYaml(metadata_yaml, route_.metadata_);
+
+  rate_limit_entry_->populateDescriptors(route_, descriptors_, "", header_, default_remote_address_,
+                                         dynamic_metadata_);
+
+  EXPECT_TRUE(descriptors_.empty());
+}
+// Tests that no descriptor is generated when both the metadata_key and default_value are empty.
+TEST_F(RateLimitPolicyEntryTest, RouteEnryMetaDataAndDefaultValueEmpty) {
+  const std::string yaml = R"EOF(
+actions:
+- route_entry_metadata:
+    descriptor_key: fake_key
+    default_value: ""
+    metadata_key:
+      key: 'envoy.xxx'
+      path:
+      - key: test
+      - key: prop
+  )EOF";
+
+  setupTest(yaml);
+
+  std::string metadata_yaml = R"EOF(
+filter_metadata:
+  envoy.xxx:
+    another_key:
+      prop: ""
+  )EOF";
+
+  TestUtility::loadFromYaml(metadata_yaml, route_.metadata_);
+
+  rate_limit_entry_->populateDescriptors(route_, descriptors_, "", header_, default_remote_address_,
+                                         dynamic_metadata_);
+
+  EXPECT_TRUE(descriptors_.empty());
+}
+
+TEST_F(RateLimitPolicyEntryTest, RouteEnryMetaDataNonStringMatch) {
+  const std::string yaml = R"EOF(
+actions:
+- route_entry_metadata:
+    descriptor_key: fake_key
+    metadata_key:
+      key: 'envoy.xxx'
+      path:
+      - key: test
+      - key: prop
+  )EOF";
+
+  setupTest(yaml);
+
+  std::string metadata_yaml = R"EOF(
+filter_metadata:
+  envoy.xxx:
+    test:
+      prop:
+        foo: bar
```

Comment:
can you add an inline comment saying that because this is a deeper map rather than a string, there is no match. Just to connect the dot between the comment above and the structure here.

Reply:
Sure and I will adjust the same testcase for dynamic metadata action. All cases are copied from there.

Reply:
I have checked the testcase name: `RouteEnryMetaDataNonStringMatch` do you still feel a comment is warranted?

Reply:
@junr03 any thoughts?

Reply:
Should it rather be RouteEnryMetaDataNonString`No`Match?

Reply:
fixed